apps/rush/src/utilities/ChangeManager.ts

```diff
@@ -0,0 +1,73 @@
+import {
+  IChangeInfo,
+  RushConfiguration,
+  RushConfigurationProject
+} from '@microsoft/rush-lib';
+
+import PublishUtilities, {
+  IChangeInfoHash
+} from './PublishUtilities';
+import ChangeFiles from './ChangeFiles';
+import PrereleaseToken from './PrereleaseToken';
+import ChangelogGenerator from './ChangelogGenerator';
+
+export default class ChangeManager {
+  private _prereleaseToken: PrereleaseToken;
+  private _orderedChanges: IChangeInfo[];
+  private _allPackages: Map<string, RushConfigurationProject>;
+  private _allChanges: IChangeInfoHash;
+  private _changeFiles: ChangeFiles;
+
+  constructor(private _rushConfiguration: RushConfiguration) {
+  }
+
+  public load(
+    changesPath: string,
+    prereleaseToken: PrereleaseToken = new PrereleaseToken(),
+    includeCommitDetails: boolean = false
+  ): void {
+    this._allPackages = this._rushConfiguration.projectsByName;
+
+    this._prereleaseToken = prereleaseToken;
+
+    this._changeFiles = new ChangeFiles(changesPath);
+    this._allChanges = PublishUtilities.findChangeRequests(
+      this._allPackages,
```

Comment:
>this._allPackages [](start = 6, length = 17)

does this really need to be its own variable if we already store rushConfiguration?